### PR TITLE
fix(listeners): deep path listeners not working bug (#219)

### LINKED ISSUE #219: [v2.0.0] bug(listeners): deep path listeners not working
**Do you want to request a *feature* or report a *bug*?**  
Bug

**What is the current behavior?**
When subscribing to multiple paths from Firebase, each reactReduxFirebase/SET action overwrites the firebase.data state entirely, wiping away any previous state from any other subscriptions.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://jsfiddle.net or similar.**
Subscribe to two paths with firebaseConnect and view the state at each reactReduxFirebase/SET action.

**What is the expected behavior?**
The data should be merged with the previous state.

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**
I noticed this when migrating my working 1.5 code to 2.0.0-beta.3.

### PR DESCRIPTION
**Pull request by marekolszewski**
### Description
Fixes #219 by using lodash's `set` one level higher in the reducer. 

### Check List

- [X] All tests passing
- [N/A] Docs updated with any changes or examples
- [N/A] Added tests to ensure feature(s) work properly

### Relevant Issues
#219

### CODE REVIEW

**Comment by prescottprue** [comment]
@marekolszewski I believe that this is an anti pattern of redux due to the not keeping [to the immutablility pattern](http://redux.js.org/docs/faq/ImmutableData.html#why-is-immutability-required).

The reason for the spread operator is to create a copy of state (is transform to the same as using `Object.assign`).

This is the code that is causing the issue, but I am hoping for a way that does not modify state.

**Comment by marekolszewski** [comment]
@prescottprue You're right, sorry, I forgot that set modifies the original object. Let me change it.

**Comment by marekolszewski** [comment]
@prescottprue OK, just updated the PR to use lodash's immutable version of set.

File: src/reducer.js [hunk 1]
```diff
@@ -149,9 +150,9 @@ export const timestampsReducer = (state = {}, { type, path }) => {
 const createDataReducer = (actionKey = 'data') => (state = {}, action) => {
   switch (action.type) {
     case SET:
-      return set(state, getDotStrPath(action.path), action[actionKey])
+      return set(getDotStrPath(action.path), action[actionKey], state)
```
Comment (prescottprue) [inline src/reducer.js]: Not sure this is what you want either. This points to the string path as the original data (which will always be a string). This set will need to be part of a spread/assign with state as it exists currently.

Comment (marekolszewski) [inline src/reducer.js]: Lodash FP functions have different argument orders from the corresponding non-FP versions (they did this to make composition easier). More details here: https://github.com/lodash/lodash/wiki/FP-Guide#rearranged-arguments

Comment (prescottprue) [inline src/reducer.js]: You are totally right! I didn't notice the `fp` in the import.

[This lodash issue confirms this usage](https://github.com/lodash/lodash/issues/1696#issuecomment-269394600). Thanks for the contribution!

**Pull request merged**